File: experiments/java/sort/files/ListII.java
interface IIterator<T>
{
	T Current();
	IIterator<T> Clone();
	boolean MoveNext();
	boolean MovePrev();
	void SetValue(T x);
}

interface IList<T>
{
	IIterator<T> GetIterator();
	long GetSize();
	void Add(T val);
}

File: experiments/higgs/sort/files/listII.ts
export module main
{
    interface IIterator
    {
	    current : () => number;
        clone : () => IIterator;
        moveNext : () => boolean;
        movePrev : () => boolean;
        setValue : (x : number) => void;
    }

    interface IList
    {
        getIterator : () => IIterator;
        getSize : () => number;
        add : (val : number) => void;
    }

